### PR TITLE
Assignment-1 

### PR DESCRIPTION
**Pull request by gaurivn**
Answers to assignment-1

### CODE REVIEW

File: JuliaGPU/session1/Gauri/q1.jl [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  x = xor(s1,s2)
+  while(x>0)
+    res = res + (x&1)
+    x = x>>>1
+  end
+  println("Hamming Distance = $res")
+end
+s1=readline()
+s2=readline()
+HammingDistance(parse(Int64,s1),parse(Int64,s2))
```
Comment (rohit-mp) [inline JuliaGPU/session1/Gauri/q1.jl]: Function name used here is wrong.

File: JuliaGPU/session1/Gauri/q2.jl [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+function CollatzSequence(N)
+    while N!=1
+        println("$N->")
+        if N%2==0
+            N = trunc(Int64,N/2)
+        else
+            N = (3*N) + 1
+
+        end
+    end
```
Comment (rohit-mp) [inline JuliaGPU/session1/Gauri/q2.jl:11]: the digit 1 is part of every Collatz sequence but is never printed here.

File: JuliaGPU/session1/Gauri/q2.jl [hunk 3]
```diff
@@ -0,0 +1,14 @@
+#2. Collatz Sequence of a no.
+function CollatzSequence(N)
+    while N!=1
+        println("$N->")
```
Comment (rohit-mp) [inline JuliaGPU/session1/Gauri/q2.jl]: the function ```print``` doesn't add a newline by default. I think that is the function you meant to use here.

File: JuliaGPU/session1/Gauri/q3.jl [hunk 4]
```diff
@@ -0,0 +1,11 @@
+#3. Printing ASCII values of each character in given string
+function printasc(str)
```
Comment (rohit-mp) [inline JuliaGPU/session1/Gauri/q3.jl:2]: the function ```printasc``` is never called.

File: JuliaGPU/session1/Gauri/q3.jl [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+        newstr = ""
+#    s = firstindex(str)
+    l = lastindex(str)
+    for i in 1:l
+        newstr*=string(Int(str[i]))
```
Comment (rohit-mp) [inline JuliaGPU/session1/Gauri/q3.jl]: This gives us the decimal representation. The questions ask for the hexadecimal representation.

**Review by rohit-mp — DISMISSED** [review]
Please fix the issues.

**Comment by gaurivn** [comment]
Okay, I'll do with print, I'll make all the changes and send again

**Comment by rohit-mp** [comment]
Make the necessary changes and commit to the same branch again. It'll reflect in the PR. No need to create another PR

**Comment by gaurivn** [comment]
Okay I'll do that, thank you for telling me, I'll only add, commit, push

**Comment by gaurivn** [comment]
I'm done pushing with all the changes

**Comment by rohit-mp** [comment]
```q2.jl``` has an error while ```q3.jl``` still doesn't print anything. Please run the programs and verify before committing.

**Comment by gaurivn** [comment]
q2 is giving load error but it prints the output, I'll make the changes and
push again in 2 minutes

File: JuliaGPU/session1/Gauri/q2.jl [hunk 6]
```diff
@@ -12,5 +11,6 @@ function CollatzSequence(N)
     end
     println("$N->")
 end
+println("Enter number")
 N=readline()
 CollatzSequence(parse(Int64,N)
```
Comment (rohit-mp) [inline JuliaGPU/session1/Gauri/q2.jl]: It's a syntax error.

**Comment by rohit-mp** [comment]
There's a syntax error in the function call to Collatz number. Fixing that should make it work

**Comment by gaurivn** [comment]
Okay I'll fix that and send

**Comment by gaurivn** [comment]
When I comment the enter the no. and function call part , it is working
fine, by just running CollatzSequence(10). I tried parsing and though I
corrected the syntax error it is still not working on atom and giving load
errors, so I'll comment that and send again.

File: JuliaGPU/session1/Gauri/q2.jl [hunk 7]
```diff
@@ -11,6 +11,6 @@ function CollatzSequence(N)
     end
     println("$N->")
 end
-println("Enter number")
-N=readline()
-CollatzSequence(parse(Int64,N)
```
Comment (rohit-mp) [inline JuliaGPU/session1/Gauri/q2.jl]: You were just missing a closing paranthesis

**Comment by gaurivn** [comment]
Yes but even with that it wasn't working, and giving load errors ( I'm
using uber-jeno editor along with Julia REPL on atom) so I commented out
all the 3 lines and began running the function by calling
CommatzSequence(10)

**Comment by gaurivn** [comment]
Please tell me should I remove the commenting #s and submit with no syntax
errors

**Comment by rohit-mp** [comment]
Adding the missing parenthesis fixed the error for me. Could you post your error here?

**Comment by gaurivn** [comment]
I had got a load error earlier .
Now I did as you said but for running, I'll have to type
CollatzSequence(10) else it won't give output.

**Comment by gaurivn** [comment]
I'll send again I'm very sorry for the inconvenience

**Comment by gaurivn** [comment]
I sent after making it how you told me to

**Comment by rohit-mp** [comment]
Are you still having any issues in running the programs?

**Comment by gaurivn** [comment]
No I'm not

**Comment by gaurivn** [comment]
Thank you

**Pull request merged**